pkg/client/connector/userd_k8s/k8s_cluster.go

```diff
@@ -170,6 +171,24 @@ func (kc *Cluster) FindAgain(c context.Context, obj kates.Object) (kates.Object,
 	return obj, nil
 }
 
+// FindPodFromDeployName returns a pod with the given name-hex-hex
+func (kc *Cluster) FindPodFromDeployName(c context.Context, namespace, name string) (*kates.Pod, error) {
+	pods, err := kc.Pods(c, namespace)
+	if err != nil {
+		return nil, err
+	}
+	regex, err := regexp.Compile(`^` + name + `-[a-z0-9]+-[a-z0-9]+$`)
+	if err != nil {
+		return nil, err
+	}
+	for _, pod := range pods {
+		if regex.MatchString(pod.Name) {
+			return pod, nil
+		}
+	}
+	return nil, errors.NewNotFound(corev1.Resource("pod"), name+"."+namespace)
```

Comment:
what about statefulsets? if a pod is called `my-sts-name-0` and the statefulset is called `my-sts-name`, the regex will not match

Instead of doing this, I suggest doing a label selector -- that should ensure things work regardless of controller (i.e. statefulset or deployment).
The workload object (i.e. `obj` in `ensureAgent`) should have a `spec.selector.matchLabels` object that you can use as a label selector to get the pods.

Reply:
oops

Reply:
obj seems to only have prebuilt functions rather than field access--there is no spec or getSpec. svc, however, has Spec, so I used that as the label selector. I do need you to look at it again to make sure im doing this right